commands/command_pre_push.go

```diff
@@ -137,7 +133,19 @@ func prePushRef(left, right string) {
 			os.Exit(2)
 		}
 	}
+}
+
+func filteredPointers(pointers []*lfs.WrappedPointer, filter lfs.StringSet) []*lfs.WrappedPointer {
```

Comment:
I tried using callbacks:

```diff
-       pointers = filteredPointers(pointers, uploadedOids)
-       pointers = filteredPointers(pointers, prePushCheckForMissingObjects(pointers))
+       filtered := make([]*lfs.WrappedPointer, 0, len(pointers))
+       cb := func(p *lfs.WrappedPointer) {
+               filtered = append(filtered, p)
+       }
+
+       filteredPointers(pointers, uploadedOids, cb)
+
+       serverOids := prePushCheckForMissingObjects(filtered)
+       filteredPointers(filtered, serverOids, cb)
```

This is more complex, and introduces a concurrency issue because the second filter call is looping through the same slice while the callback is appending to it. The original approach, though 30% more wasteful with allocations, is the best option IMO. Though, that's assuming filtering twice is a good idea. My intention is to filter out everything that has been uploaded from this same process before checking the server to see what it has. It'll result in fewer batch calls in large import pushes.